### PR TITLE
Better support of GCP storage

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp [hunk 1]
```diff
@@ -283,64 +285,31 @@ void S3ObjectStorage::removeObjects(const std::vector<std::string> & paths)
             request.SetBucket(bucket);
             request.SetDelete(delkeys);
             auto outcome = client_ptr->DeleteObjects(request);
-            throwIfError(outcome);
+
+            if (!outcome.IsSuccess() && (!if_exists || !isNotFoundError(outcome.GetError().GetErrorType())))
+                throwIfError(outcome);
```
Comment (CurtizJ) [inline src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:290]: Also, in this case even if absence of key may be ok for us, the log will be polluted with error messages from aws sdk. I don't know how to supress them.

**Comment by CurtizJ** [comment]
@Mergifyio update

**Review by alesapin — APPROVED** [review]

File: src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp [hunk 2]
```diff
@@ -255,7 +257,7 @@ void S3ObjectStorage::removeObjects(const std::vector<std::string> & paths)
     if (!s3_capabilities.support_batch_delete)
     {
         for (const auto & path : paths)
-            removeObject(path);
+            removeObjectImpl(path, if_exists);
```
Comment (rschu1ze) [inline src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:260]: GitHub somehow won't let me add a comment on an unchanged line and it's kindof unrelated, but the scope of the variables in l. 254/255 could be decreased.

File: src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp (hunk 1 above)
Comment (rschu1ze) [reply src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:290]: It may also make sense to have l. 289-290 resp. 217-218 as a separate function.

**Review by rschu1ze — COMMENTED** [review]
Just two smaller comments besides Sashas comments.

**Review by rschu1ze — APPROVED** [review]

**Comment by alesapin** [comment]
```
test_storage_s3/test.py::test_seekable_formats_url (✕2)
``` 
Flaky now?

**Comment by CurtizJ** [comment]
Stateless tests (release, s3 storage, actions) : `SystemLog (system.opentelemetry_span_log): Queue is full for system log 'DB::OpenTelemetrySpanLog' at 1944719` is flaky in s3 check
example with another system table: https://s3.amazonaws.com/clickhouse-test-reports/0/7cc87d9a65572ba06206d06f770929a7ba245d6f/stateless_tests__release__s3_storage__actions_.html

AST Fuzzer and Stress Test are unrelated.

**Pull request merged**